lib/plugins/aws/deploy/compile/functions/index.js

```diff
@@ -164,8 +165,26 @@ class AwsCompileFunctions {
         delete newFunction.Properties.VpcConfig;
       }
 
+      // add the VPC permissions to the lambda execution role if a VPC config is given
+      if (newFunction.Properties.VpcConfig) {
+        // merge in the iamPolicyLambdaVPCTemplate
+        const iamPolicyLambdaVpcTemplate = this.serverless.utils.readFileSync(
+          path.join(this.serverless.config.serverlessPath,
+            'plugins',
+            'aws',
+            'deploy',
+            'compile',
+            'functions',
+            'iam-policy-lambda-vpc-template.json')
+        );
+
+        _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
+          iamPolicyLambdaVpcTemplate);
```

Comment:
If iamRoleARN is defined on provider (so as to bypass the use of the default role/policy) then the role IamRoleLambda won't exist (see https://github.com/serverless/serverless/blob/master/lib/plugins/aws/deploy/compile/functions/index.js#L18) and this merge probably shouldn't be made.
(see also PR #2073 which is expanding the ability not to use the default role and policy IAM approach)
FWIW: Adding a policy to the CFT will block us from successfully deploying the CFT and thus from using SLS.